partials/index/Request.njk

```diff
@@ -0,0 +1,29 @@
+
+{% macro request(defaultContentType, channelName, publishMessage, subscribeMessage, messageDescription, channelParameters) %}
+  /**
+  *  {{messageDescription}}
+  * @param requestMessage The request message to send.
+  */
+  public request{{channelName | pascalCase }}(requestMessage: {{ publishMessage.uid() | pascalCase }}Message 
+  {% if channelParameters | length %}
+    ,{{channelParameters | realizeParametersForChannel}}
+  {% endif %}): Promise<{{ subscribeMessage.uid() | pascalCase }}Message> {
+    {%- if publishMessage.contentType() | isBinaryPayload(defaultContentType) %}
+    const nc: Client = this.binaryClient!;
+    {%- endif %}
+    {%- if publishMessage.contentType() | isStringPayload(defaultContentType) %}
+    const nc: Client = this.stringClient!;
+    {%- endif %}
+    {%- if publishMessage.contentType() | isJsonPayload(defaultContentType) %}
+    const nc: Client = this.jsonClient!;
```

Comment:
I am actually contemplating allowing the nats client to simply use binary payloads. And have the client decide to decode/encode itself. Part of the problem is that for example, if you have a JSON client, the API requires that everything be valid JSON.

If the decoding of the payload is a function on the message, it becomes possible to let the subscriber/requestor to decide how to parse it back. This would simplify the client a bit, at the expense of some changes in the publisher.

Reply:
Interesting, it definitely took me off guard when first using the client that you had to have a client for each of the three protocols (compared to other client implementations such as the official  .NET client).